float64 left
float64 center
float64 right
